Binary files qt-2.3.7/bin/moc and qte-2.3.7/bin/moc differ
Binary files qt-2.3.7/bin/uic and qte-2.3.7/bin/uic differ
diff -uNr --exclude-from=./diff_ignore qt-2.3.7/configs/darwin-generic-g++-shared qte-2.3.7/configs/darwin-generic-g++-shared
--- qt-2.3.7/configs/darwin-generic-g++-shared	Thu Jan  1 01:00:00 1970
+++ qte-2.3.7/configs/darwin-generic-g++-shared	Mon Oct 27 14:27:25 2003
@@ -0,0 +1,98 @@
+# Compiling
+INTERFACE_DECL_PATH 	= .
+SYSCONF_CXX		= g++
+SYSCONF_CC		= gcc
+DASHCROSS		= 
+
+# Compiling with support libraries
+SYSCONF_CXXFLAGS_X11	= -I/usr/X11R6/include
+SYSCONF_CXXFLAGS_QT	= -I$(QTDIR)/include
+SYSCONF_CXXFLAGS_QTOPIA	= -I$(QPEDIR)/include
+SYSCONF_CXXFLAGS_OPENGL	= -I/usr/X11R6/include
+
+# Compiling YACC output
+SYSCONF_CXXFLAGS_YACC     = -Wno-unused -Wno-parentheses
+
+# Linking with support libraries
+SYSCONF_RPATH_X11	= -L/usr/X11R6/lib
+SYSCONF_RPATH_QT	= -L$(QTDIR)/lib
+SYSCONF_RPATH_QTOPIA	= -L$(QPEDIR)/lib
+SYSCONF_RPATH_OPENGL	= -L/usr/X11R6/lib
+
+# Linking with support libraries
+# X11
+SYSCONF_LFLAGS_X11	= -L/usr/X11R6/lib
+SYSCONF_LIBS_X11	= 
+# Qt, Qt+OpenGL
+SYSCONF_LFLAGS_QT	= -L$(QTDIR)/lib
+SYSCONF_LFLAGS_QTOPIA	= -L$(QPEDIR)/lib
+SYSCONF_LIBS_QT		= -lqte$(QT_THREAD_SUFFIX)
+SYSCONF_LIBS_QT_OPENGL	= 
+SYSCONF_LIBS_QTOPIA	= -lqtopia
+# OpenGL
+SYSCONF_LFLAGS_OPENGL	= -L/usr/X11R6/lib
+SYSCONF_LIBS_OPENGL	= -lGLU -lGL -lXmu
+# Yacc
+SYSCONF_LIBS_YACC	= 
+
+# Linking applications
+SYSCONF_LINK		= g++
+SYSCONF_LFLAGS		= 
+SYSCONF_LIBS		= 
+
+# Link flags for shared objects
+SYSCONF_LFLAGS_SHOBJ	= -dynamiclib
+
+# Flags for threading
+SYSCONF_CFLAGS_THREAD	= -pthread -D_THREAD_SAFE
+SYSCONF_CXXFLAGS_THREAD	= -pthread -D_THREAD_SAFE
+SYSCONF_LFLAGS_THREAD	= -pthread
+SYSCONF_LIBS_THREAD	= 
+
+# Meta-object compiler
+SYSCONF_MOC		= $(QTDIR)/bin/moc
+
+# UI compiler
+SYSCONF_UIC		= $(QTDIR)/bin/uic
+
+# Linking shared libraries
+#   - Build the $(TARGET) library, eg. lib$(TARGET).so.2.2.2
+#   - Place target in $(DESTDIR) - which has a trailing /
+#   - Usually needs to incorporate $(VER_MAJ), $(VER_MIN) and $(VER_PATCH)
+#
+SYSCONF_LINK_SHLIB	= g++
+SYSCONF_LINK_TARGET_SHARED	= lib$(TARGET).dylib.$(VER_MAJ).$(VER_MIN).$(VER_PATCH)
+SYSCONF_LINK_LIB_SHARED	=  $(SYSCONF_LINK_SHLIB) -dynamiclib  \
+				     $(LFLAGS) -o $(SYSCONF_LINK_TARGET_SHARED) \
+				     $(OBJECTS) $(OBJMOC) $(LIBS) && \
+				 mv $(SYSCONF_LINK_TARGET_SHARED) $(DESTDIR); \
+				 cd $(DESTDIR) && \
+				 rm -f lib$(TARGET).dylib lib$(TARGET).dylib.$(VER_MAJ) lib$(TARGET).dylib.$(VER_MAJ).$(VER_MIN); \
+				 ln -s $(SYSCONF_LINK_TARGET_SHARED) lib$(TARGET).dylib; \
+				 ln -s $(SYSCONF_LINK_TARGET_SHARED) lib$(TARGET).dylib.$(VER_MAJ); \
+				 ln -s $(SYSCONF_LINK_TARGET_SHARED) lib$(TARGET).dylib.$(VER_MAJ).$(VER_MIN)
+
+# Linking static libraries
+#   - Build the $(TARGET) library, eg. lib$(TARGET).a
+#   - Place target in $(DESTDIR) - which has a trailing /
+#
+SYSCONF_AR		= ar cqs
+SYSCONF_LINK_TARGET_STATIC = lib$(TARGET).a
+SYSCONF_LINK_LIB_STATIC	= rm -f $(DESTDIR)$(SYSCONF_LINK_TARGET_STATIC) ; \
+				 $(SYSCONF_AR) $(DESTDIR)$(SYSCONF_LINK_TARGET_STATIC) $(OBJECTS) $(OBJMOC) 
+# Compiling application source
+SYSCONF_CXXFLAGS	= -pipe -fno-exceptions -DQWS -fno-rtti -fno-exceptions -DQT_NO_QWS_LINUXFB -DQT_NO_QWS_TRANSFORMED -O0 -Wall -W -DNO_DEBUG
+SYSCONF_CFLAGS		= -pipe -fno-exceptions -O2 -Wall -W
+# Default link type (static linking is still be used where required)
+SYSCONF_LINK_LIB	= $(SYSCONF_LINK_LIB_SHARED)
+SYSCONF_LINK_TARGET	= $(SYSCONF_LINK_TARGET_SHARED)
+# Compiling library source
+SYSCONF_CXXFLAGS_LIB	= -fPIC
+SYSCONF_CFLAGS_LIB	= -fPIC
+# Compiling shared-object source
+SYSCONF_CXXFLAGS_SHOBJ	= -fPIC
+SYSCONF_CFLAGS_SHOBJ	= -fPIC
+# Linking Qt
+SYSCONF_LIBS_QTLIB	= $(SYSCONF_LFLAGS_X11) $(QT_LIBS_MT) $(QT_LIBS_OPT)
+# Linking Qt applications
+SYSCONF_LIBS_QTAPP	= 
diff -uNr --exclude-from=./diff_ignore qt-2.3.7/include/qapplication.h qte-2.3.7/include/qapplication.h
--- qt-2.3.7/include/qapplication.h	Thu Jul 17 03:20:25 2003
+++ qte-2.3.7/include/qapplication.h	Mon Oct 27 14:21:28 2003
@@ -61,6 +61,10 @@
 class QSemaphore;
 #endif
 
+#if !defined( QT_WEAK_SYMBOL )
+#define QT_WEAK_SYMBOL
+#endif
+
 // REMOVE IN 3.0 (just here for moc source compatibility)
 #define QNonBaseApplication QApplication
 
@@ -85,7 +89,10 @@
 
 #ifndef QT_NO_STYLE
     static QStyle  &style();
-    static void	    setStyle( QStyle* );
+    static void	    setStyle( QStyle* ) QT_WEAK_SYMBOL;
+private:
+	static void     setStyle_NonWeak( QStyle* );
+public:
 #endif
 #if 1	/* OBSOLETE */
     enum ColorMode { NormalColors, CustomColors };
@@ -106,11 +113,19 @@
 #ifndef QT_NO_PALETTE
     static QPalette  palette( const QWidget* = 0 );
     static void	     setPalette( const QPalette &, bool informWidgets=FALSE,
+				 const char* className = 0 ) QT_WEAK_SYMBOL;
+private:
+    static void	     setPalette_NonWeak( const QPalette &, bool informWidgets=FALSE,
 				 const char* className = 0 );
+public:
 #endif
     static QFont     font( const QWidget* = 0 );
     static void	     setFont( const QFont &, bool informWidgets=FALSE,
+			      const char* className = 0 ) QT_WEAK_SYMBOL;
+private:
+    static void	     setFont_NonWeak( const QFont &, bool informWidgets=FALSE,
 			      const char* className = 0 );
+public:			      
     static QFontMetrics fontMetrics();
 
     QWidget	    *mainWidget()  const;
@@ -207,7 +222,10 @@
     void             qwsSetCustomColors( QRgb *colortable, int start, int numColors );
 #ifndef QT_NO_QWS_MANAGER
     static QWSDecoration &qwsDecoration();
-    static void      qwsSetDecoration( QWSDecoration *);
+    static void      qwsSetDecoration( QWSDecoration *) QT_WEAK_SYMBOL;
+private:
+    static void      qwsSetDecoration_NonWeak( QWSDecoration *);
+public:
 #endif
 #endif
 
diff -uNr --exclude-from=./diff_ignore qt-2.3.7/include/qconfig-qpe.h qte-2.3.7/include/qconfig-qpe.h
--- qt-2.3.7/include/qconfig-qpe.h	Thu Jan  1 01:00:00 1970
+++ qte-2.3.7/include/qconfig-qpe.h	Sun Dec 14 12:48:43 2003
@@ -0,0 +1,100 @@
+/**********************************************************************
+** Copyright (C) 2000 Trolltech AS.  All rights reserved.
+**
+** This file is part of Qtopia Environment.
+**
+** This file may be distributed and/or modified under the terms of the
+** GNU General Public License version 2 as published by the Free Software
+** Foundation and appearing in the file LICENSE.GPL included in the
+** packaging of this file.
+**
+** This file is provided AS IS with NO WARRANTY OF ANY KIND, INCLUDING THE
+** WARRANTY OF DESIGN, MERCHANTABILITY AND FITNESS FOR A PARTICULAR PURPOSE.
+**
+** See http://www.trolltech.com/gpl/ for GPL licensing information.
+**
+** Contact info@trolltech.com if any conditions of this licensing are
+** not clear to you.
+**
+**********************************************************************/
+#ifndef QT_H
+#endif // QT_H
+
+// Empty leaves all features enabled.  See doc/html/features.html for choices.
+
+// Note that disabling some features will produce a libqt that is not
+// compatible with other libqt builds. Such modifications are only
+// supported on Qt/Embedded where reducing the library size is important
+// and where the application-suite is often a fixed set.
+
+#ifndef QT_DLL
+#define QT_DLL // Internal
+#endif
+
+#if defined(QT_QWS_IPAQ) || defined(QT_QWS_CASSIOPEIA) || defined(QT_QWS_SL5XXX)
+# define QT_NO_QWS_CURSOR
+# define QT_NO_QWS_MOUSE_AUTO
+#endif
+#ifndef QT_NO_CODECS
+#define QT_NO_CODECS
+#endif
+#define QT_NO_UNICODETABLES
+//#define QT_NO_IMAGEIO_BMP
+#define QT_NO_IMAGEIO_PPM
+//#define QT_NO_ASYNC_IO
+//#define QT_NO_ASYNC_IMAGE_IO
+#define QT_NO_FREETYPE
+#define QT_NO_BDF
+//#define QT_NO_FONTDATABASE
+#define QT_NO_DRAGANDDROP
+//#define QT_NO_CLIPBOARD
+#define QT_NO_PROPERTIES
+#define QT_NO_NETWORKPROTOCOL
+
+#define QT_NO_IMAGE_TEXT
+
+//#define QT_NO_TOOLTIP
+#define QT_NO_COLORNAMES
+#define QT_NO_TRANSFORMATIONS
+#define QT_NO_TRANSLATION_BUILDER
+#define QT_NO_COMPLEXTEXT
+#define QT_NO_PRINTER
+#define QT_NO_PICTURE
+//#define QT_NO_ICONVIEW
+#define QT_NO_DIAL
+#define QT_NO_SIZEGRIP
+#define QT_NO_WORKSPACE
+//#define QT_NO_TABLE
+//#define QT_NO_ACTION
+//#define QT_NO_SETTINGS
+#define QT_NO_STYLE_POCKETPC
+#ifndef QT_NO_STYLE_AQUA
+# define QT_NO_STYLE_AQUA
+#endif
+#define QT_NO_STYLE_MOTIF
+#define QT_NO_STYLE_PLATINUM
+#define QT_NO_FILEDIALOG
+#define QT_NO_FONTDIALOG
+#define QT_NO_PRINTDIALOG
+#define QT_NO_COLORDIALOG
+#define QT_NO_INPUTDIALOG
+//#define QT_NO_MESSAGEBOX
+#define QT_NO_PROGRESSDIALOG
+//#define QT_NO_TABDIALOG
+#define QT_NO_WIZARD
+#define QT_NO_EFFECTS
+//#define QT_NO_COMPONENT
+#define QT_NO_DOM
+#define QT_NO_SEMIMODAL
+//#define QT_NO_PROGRESSBAR
+#define QT_NO_SPLITTER
+
+//#define QT_NO_QWS_SAVEFONTS
+//#define QT_NO_QWS_PROPERTIES
+
+#define QT_NO_QWS_BEOS_WM_STYLE
+#define QT_NO_QWS_KDE2_WM_STYLE
+#define QT_NO_QWS_KDE_WM_STYLE
+#define QT_NO_QWS_WINDOWS_WM_STYLE
+
+#define QT_NO_SOUND
diff -uNr --exclude-from=./diff_ignore qt-2.3.7/include/qconfig.h qte-2.3.7/include/qconfig.h
--- qt-2.3.7/include/qconfig.h	Fri Jan 26 14:43:06 2001
+++ qte-2.3.7/include/qconfig.h	Thu Dec 11 12:54:20 2003
@@ -1,3 +1,26 @@
+// Copied from qconfig-qpe.h
+/**********************************************************************
+** Copyright (C) 2000 Trolltech AS.  All rights reserved.
+**
+** This file is part of Qtopia Environment.
+**
+** This file may be distributed and/or modified under the terms of the
+** GNU General Public License version 2 as published by the Free Software
+** Foundation and appearing in the file LICENSE.GPL included in the
+** packaging of this file.
+**
+** This file is provided AS IS with NO WARRANTY OF ANY KIND, INCLUDING THE
+** WARRANTY OF DESIGN, MERCHANTABILITY AND FITNESS FOR A PARTICULAR PURPOSE.
+**
+** See http://www.trolltech.com/gpl/ for GPL licensing information.
+**
+** Contact info@trolltech.com if any conditions of this licensing are
+** not clear to you.
+**
+**********************************************************************/
+#ifndef QT_H
+#endif // QT_H
+
 // Empty leaves all features enabled.  See doc/html/features.html for choices.
 
 // Note that disabling some features will produce a libqt that is not
@@ -8,3 +31,71 @@
 #ifndef QT_DLL
 #define QT_DLL // Internal
 #endif
+
+#if defined(QT_QWS_IPAQ) || defined(QT_QWS_CASSIOPEIA) || defined(QT_QWS_SL5XXX)
+# define QT_NO_QWS_CURSOR
+# define QT_NO_QWS_MOUSE_AUTO
+#endif
+#ifndef QT_NO_CODECS
+#define QT_NO_CODECS
+#endif
+#define QT_NO_UNICODETABLES
+//#define QT_NO_IMAGEIO_BMP
+#define QT_NO_IMAGEIO_PPM
+//#define QT_NO_ASYNC_IO
+//#define QT_NO_ASYNC_IMAGE_IO
+#define QT_NO_FREETYPE
+#define QT_NO_BDF
+//#define QT_NO_FONTDATABASE
+#define QT_NO_DRAGANDDROP
+//#define QT_NO_CLIPBOARD
+#define QT_NO_PROPERTIES
+#define QT_NO_NETWORKPROTOCOL
+
+#define QT_NO_IMAGE_TEXT
+
+//#define QT_NO_TOOLTIP
+#define QT_NO_COLORNAMES
+#define QT_NO_TRANSFORMATIONS
+#define QT_NO_TRANSLATION_BUILDER
+#define QT_NO_COMPLEXTEXT
+#define QT_NO_PRINTER
+#define QT_NO_PICTURE
+//#define QT_NO_ICONVIEW
+#define QT_NO_DIAL
+#define QT_NO_SIZEGRIP
+#define QT_NO_WORKSPACE
+//#define QT_NO_TABLE
+//#define QT_NO_ACTION
+//#define QT_NO_SETTINGS
+#define QT_NO_STYLE_POCKETPC
+#ifndef QT_NO_STYLE_AQUA
+# define QT_NO_STYLE_AQUA
+#endif
+#define QT_NO_STYLE_MOTIF
+#define QT_NO_STYLE_PLATINUM
+#define QT_NO_FILEDIALOG
+#define QT_NO_FONTDIALOG
+#define QT_NO_PRINTDIALOG
+#define QT_NO_COLORDIALOG
+#define QT_NO_INPUTDIALOG
+//#define QT_NO_MESSAGEBOX
+#define QT_NO_PROGRESSDIALOG
+//#define QT_NO_TABDIALOG
+#define QT_NO_WIZARD
+#define QT_NO_EFFECTS
+//#define QT_NO_COMPONENT
+#define QT_NO_DOM
+#define QT_NO_SEMIMODAL
+//#define QT_NO_PROGRESSBAR
+#define QT_NO_SPLITTER
+
+//#define QT_NO_QWS_SAVEFONTS
+//#define QT_NO_QWS_PROPERTIES
+
+#define QT_NO_QWS_BEOS_WM_STYLE
+#define QT_NO_QWS_KDE2_WM_STYLE
+#define QT_NO_QWS_KDE_WM_STYLE
+#define QT_NO_QWS_WINDOWS_WM_STYLE
+
+#define QT_NO_SOUND
\ No newline at end of file
diff -uNr --exclude-from=./diff_ignore qt-2.3.7/include/qfontdatabase.h qte-2.3.7/include/qfontdatabase.h
--- qt-2.3.7/include/qfontdatabase.h	Thu Jul 17 03:20:25 2003
+++ qte-2.3.7/include/qfontdatabase.h	Mon Oct 27 14:21:28 2003
@@ -59,6 +59,10 @@
 class QDiskFont;
 #endif
 
+#if !defined( QT_WEAK_SYMBOL )
+#define QT_WEAK_SYMBOL
+#endif
+
 class QFontDatabasePrivate;
 
 class Q_EXPORT QFontDatabase
@@ -67,9 +71,16 @@
     QFontDatabase();
 
     QStringList families( bool onlyForLocale = TRUE ) const;
+    
+    
     QValueList<int> pointSizes( const QString &family,
 				const QString &style = QString::null,
-				const QString &charSet = QString::null );
+				const QString &charSet = QString::null ) QT_WEAK_SYMBOL;
+private:
+    QValueList<int> pointSizes_NonWeak( const QString &family,
+				const QString &style,
+				const QString &charSet );
+public:
     QStringList styles( const QString &family,
 			const QString &charSet = QString::null ) const;
     QStringList charSets( const QString &familyName,
diff -uNr --exclude-from=./diff_ignore qt-2.3.7/include/qgifimageformat_p.h qte-2.3.7/include/qgifimageformat_p.h
--- qt-2.3.7/include/qgifimageformat_p.h	Thu Jul 17 03:20:24 2003
+++ qte-2.3.7/include/qgifimageformat_p.h	Thu Jan  1 01:00:00 1970
@@ -1,56 +0,0 @@
-/****************************************************************************
-** $Id: qte237-all-macosx.patch,v 1.3 2003-12-15 09:42:43 eilers Exp $
-**
-** Implementation of gif image/movie loading classes
-**
-** Created : 970617
-**
-** Copyright (C) 1992-2000 Trolltech AS.  All rights reserved.
-**
-** This file is part of the kernel module of the Qt GUI Toolkit.
-**
-** This file may be distributed under the terms of the Q Public License
-** as defined by Trolltech AS of Norway and appearing in the file
-** LICENSE.QPL included in the packaging of this file.
-**
-** This file may be distributed and/or modified under the terms of the
-** GNU General Public License version 2 as published by the Free Software
-** Foundation and appearing in the file LICENSE.GPL included in the
-** packaging of this file.
-**
-** Licensees holding valid Qt Enterprise Edition or Qt Professional Edition
-** licenses may use this file in accordance with the Qt Commercial License
-** Agreement provided with the Software.
-**
-** This file is provided AS IS with NO WARRANTY OF ANY KIND, INCLUDING THE
-** WARRANTY OF DESIGN, MERCHANTABILITY AND FITNESS FOR A PARTICULAR PURPOSE.
-**
-** See http://www.trolltech.com/pricing.html or email sales@trolltech.com for
-**   information about Qt Commercial License Agreements.
-** See http://www.trolltech.com/qpl/ for QPL licensing information.
-** See http://www.trolltech.com/gpl/ for GPL licensing information.
-**
-** Contact info@trolltech.com if any conditions of this licensing are
-** not clear to you.
-**
-**********************************************************************/
-
-#ifndef QT_H
-#include "qasyncimageio.h"
-#include "qgif.h"
-#endif // QT_H
-
-#ifndef QT_NO_ASYNC_IMAGE_IO
-
-// See qgif.h for important information regarding this option
-#if defined(QT_BUILTIN_GIF_READER) && QT_BUILTIN_GIF_READER == 1
-
-class Q_EXPORT QGIFFormatType : public QImageFormatType
-{
-    QImageFormat* decoderFor(const uchar* buffer, int length);
-    const char* formatName() const;
-};
-
-#endif // QT_BUILTIN_GIF_READER
-
-#endif // QT_NO_ASYNC_IMAGE_IO
diff -uNr --exclude-from=./diff_ignore qt-2.3.7/include/qmodules.h qte-2.3.7/include/qmodules.h
--- qt-2.3.7/include/qmodules.h	Fri Aug 31 17:00:56 2001
+++ qte-2.3.7/include/qmodules.h	Thu Dec 11 12:53:57 2003
@@ -1,14 +1,12 @@
-#ifndef QT_H
-#endif // QT_H
-
+// These modules are licensed to you
 #define QT_MODULE_TOOLS
 #define QT_MODULE_KERNEL
 #define QT_MODULE_WIDGETS
 #define QT_MODULE_DIALOGS
-
 #define QT_MODULE_ICONVIEW
 #define QT_MODULE_WORKSPACE
-#define QT_MODULE_TABLE
+#define QT_MODULE_NETWORK
 #define QT_MODULE_CANVAS
+#define QT_MODULE_TABLE
 #define QT_MODULE_XML
-#define QT_MODULE_NETWORK
+#define QT_MODULE_OPENGL
Binary files qt-2.3.7/lib/libqte.dylib and qte-2.3.7/lib/libqte.dylib differ
Binary files qt-2.3.7/lib/libqte.dylib.2 and qte-2.3.7/lib/libqte.dylib.2 differ
Binary files qt-2.3.7/lib/libqte.dylib.2.3 and qte-2.3.7/lib/libqte.dylib.2.3 differ
Binary files qt-2.3.7/lib/libqte.dylib.2.3.7 and qte-2.3.7/lib/libqte.dylib.2.3.7 differ
diff -uNr --exclude-from=./diff_ignore qt-2.3.7/src/kernel/qapplication.cpp qte-2.3.7/src/kernel/qapplication.cpp
--- qt-2.3.7/src/kernel/qapplication.cpp	Thu Jul 17 03:20:25 2003
+++ qte-2.3.7/src/kernel/qapplication.cpp	Mon Oct 27 14:21:28 2003
@@ -35,6 +35,8 @@
 **
 **********************************************************************/
 
+#define QT_WEAK_SYMBOL	__attribute__(( weak ))
+
 #include "qobjectlist.h"
 #include "qobjectdict.h"
 #include "qapplication.h"
@@ -933,11 +935,16 @@
 #ifndef QT_NO_STYLE
 void QApplication::setStyle( QStyle *style )
 {
+	setStyle_NonWeak ( style );
+}
+
+void QApplication::setStyle_NonWeak( QStyle *style )
+{
     QStyle* old = app_style;
-    app_style = style;
 
     if ( startingUp() ) {
 	delete old;
+	app_style = style;
 	return;
     }
 
@@ -958,6 +965,8 @@
 	old->unPolish( qApp );
     }
 
+    app_style = style;
+ 
     // take care of possible palette requirements of certain gui
     // styles. Do it before polishing the application since the style
     // might call QApplication::setStyle() itself
@@ -1184,13 +1193,30 @@
   \sa QWidget::setPalette(), palette(), QStyle::polish()
 */
 
-void QApplication::setPalette( const QPalette &palette, bool informWidgets,
+void QApplication::setPalette ( const QPalette &palette, bool informWidgets,
+			       const char* className )
+{
+	setPalette_NonWeak ( palette, informWidgets, className );
+}
+
+void QApplication::setPalette_NonWeak ( const QPalette &palette, bool informWidgets,
 			       const char* className )
 {
     QPalette pal = palette;
 #ifndef QT_NO_STYLE
-    if ( !startingUp() )
+    if ( !startingUp() ) {
 	qApp->style().polish( pal );	// NB: non-const reference
+	if ( className ) {
+	    // if we just polished a class specific palette (this normally 
+	    // only called by qt_fix_tooltips - see below), we better re-
+	    // polish the global palette. Some styles like liquid can get 
+	    // confused, because they can not detect if the polished palette 
+	    // is the global one or only a class specific one.
+	    // (liquid uses this palette to calculate blending pixmaps)
+	    QPalette p = qApp-> palette ( );
+	    qApp->style().polish ( p );
+	}
+    }
 #endif
     bool all = FALSE;
     if ( !className ) {
@@ -1273,6 +1299,12 @@
 */
 
 void QApplication::setFont( const QFont &font, bool informWidgets,
+			    const char* className )
+{
+	setFont_NonWeak ( font, informWidgets, className );
+}
+
+void QApplication::setFont_NonWeak( const QFont &font, bool informWidgets,
 			    const char* className )
 {
     bool all = FALSE;
diff -uNr --exclude-from=./diff_ignore qt-2.3.7/src/kernel/qapplication.h qte-2.3.7/src/kernel/qapplication.h
--- qt-2.3.7/src/kernel/qapplication.h	Thu Jul 17 03:20:25 2003
+++ qte-2.3.7/src/kernel/qapplication.h	Mon Oct 27 14:21:28 2003
@@ -61,6 +61,10 @@
 class QSemaphore;
 #endif
 
+#if !defined( QT_WEAK_SYMBOL )
+#define QT_WEAK_SYMBOL
+#endif
+
 // REMOVE IN 3.0 (just here for moc source compatibility)
 #define QNonBaseApplication QApplication
 
@@ -85,7 +89,10 @@
 
 #ifndef QT_NO_STYLE
     static QStyle  &style();
-    static void	    setStyle( QStyle* );
+    static void	    setStyle( QStyle* ) QT_WEAK_SYMBOL;
+private:
+	static void     setStyle_NonWeak( QStyle* );
+public:
 #endif
 #if 1	/* OBSOLETE */
     enum ColorMode { NormalColors, CustomColors };
@@ -106,11 +113,19 @@
 #ifndef QT_NO_PALETTE
     static QPalette  palette( const QWidget* = 0 );
     static void	     setPalette( const QPalette &, bool informWidgets=FALSE,
+				 const char* className = 0 ) QT_WEAK_SYMBOL;
+private:
+    static void	     setPalette_NonWeak( const QPalette &, bool informWidgets=FALSE,
 				 const char* className = 0 );
+public:
 #endif
     static QFont     font( const QWidget* = 0 );
     static void	     setFont( const QFont &, bool informWidgets=FALSE,
+			      const char* className = 0 ) QT_WEAK_SYMBOL;
+private:
+    static void	     setFont_NonWeak( const QFont &, bool informWidgets=FALSE,
 			      const char* className = 0 );
+public:			      
     static QFontMetrics fontMetrics();
 
     QWidget	    *mainWidget()  const;
@@ -207,7 +222,10 @@
     void             qwsSetCustomColors( QRgb *colortable, int start, int numColors );
 #ifndef QT_NO_QWS_MANAGER
     static QWSDecoration &qwsDecoration();
-    static void      qwsSetDecoration( QWSDecoration *);
+    static void      qwsSetDecoration( QWSDecoration *) QT_WEAK_SYMBOL;
+private:
+    static void      qwsSetDecoration_NonWeak( QWSDecoration *);
+public:
 #endif
 #endif
 
diff -uNr --exclude-from=./diff_ignore qt-2.3.7/src/kernel/qapplication_qws.cpp qte-2.3.7/src/kernel/qapplication_qws.cpp
--- qt-2.3.7/src/kernel/qapplication_qws.cpp	Thu Jul 17 03:20:25 2003
+++ qte-2.3.7/src/kernel/qapplication_qws.cpp	Mon Oct 27 14:21:28 2003
@@ -2804,6 +2804,11 @@
 */
 void QApplication::qwsSetDecoration( QWSDecoration *d )
 {
+	qwsSetDecoration_NonWeak ( d );
+}
+
+void QApplication::qwsSetDecoration_NonWeak( QWSDecoration *d )
+{
     if ( d ) {
 	delete qws_decoration;
 	qws_decoration = d;
diff -uNr --exclude-from=./diff_ignore qt-2.3.7/src/kernel/qfontdatabase.cpp qte-2.3.7/src/kernel/qfontdatabase.cpp
--- qt-2.3.7/src/kernel/qfontdatabase.cpp	Thu Jul 17 03:20:25 2003
+++ qte-2.3.7/src/kernel/qfontdatabase.cpp	Mon Oct 27 14:21:28 2003
@@ -35,6 +35,8 @@
 **
 **********************************************************************/
 
+#define QT_WEAK_SYMBOL  __attribute__(( weak ))
+
 #include "qfontdatabase.h"
 
 #ifndef QT_NO_FONTDATABASE
@@ -2421,6 +2423,13 @@
 */
 
 QValueList<int> QFontDatabase::pointSizes( const QString &family,
+                                                 const QString &style,
+                                                 const QString &charSet )
+{
+	return pointSizes_NonWeak ( family, style, charSet );
+}
+
+QValueList<int> QFontDatabase::pointSizes_NonWeak ( const QString &family,
                                                  const QString &style,
                                                  const QString &charSet )
 {
diff -uNr --exclude-from=./diff_ignore qt-2.3.7/src/kernel/qfontdatabase.h qte-2.3.7/src/kernel/qfontdatabase.h
--- qt-2.3.7/src/kernel/qfontdatabase.h	Thu Jul 17 03:20:25 2003
+++ qte-2.3.7/src/kernel/qfontdatabase.h	Mon Oct 27 14:21:28 2003
@@ -59,6 +59,10 @@
 class QDiskFont;
 #endif
 
+#if !defined( QT_WEAK_SYMBOL )
+#define QT_WEAK_SYMBOL
+#endif
+
 class QFontDatabasePrivate;
 
 class Q_EXPORT QFontDatabase
@@ -67,9 +71,16 @@
     QFontDatabase();
 
     QStringList families( bool onlyForLocale = TRUE ) const;
+    
+    
     QValueList<int> pointSizes( const QString &family,
 				const QString &style = QString::null,
-				const QString &charSet = QString::null );
+				const QString &charSet = QString::null ) QT_WEAK_SYMBOL;
+private:
+    QValueList<int> pointSizes_NonWeak( const QString &family,
+				const QString &style,
+				const QString &charSet );
+public:
     QStringList styles( const QString &family,
 			const QString &charSet = QString::null ) const;
     QStringList charSets( const QString &familyName,
diff -uNr --exclude-from=./diff_ignore qt-2.3.7/src/kernel/qgfx_qws.cpp qte-2.3.7/src/kernel/qgfx_qws.cpp
--- qt-2.3.7/src/kernel/qgfx_qws.cpp	Thu Jul 17 03:20:25 2003
+++ qte-2.3.7/src/kernel/qgfx_qws.cpp	Mon Oct 27 14:27:25 2003
@@ -32,7 +32,7 @@
 #include "qgfx_qws.h"
 
 #include <stdio.h>
-#include <dlfcn.h>
+//#include <dlfcn.h>
 #include <sys/types.h>
 #include <dirent.h>
 #include <stdlib.h>
diff -uNr --exclude-from=./diff_ignore qt-2.3.7/src/kernel/qgfxraster_qws.cpp qte-2.3.7/src/kernel/qgfxraster_qws.cpp
--- qt-2.3.7/src/kernel/qgfxraster_qws.cpp	Thu Jul 17 03:20:25 2003
+++ qte-2.3.7/src/kernel/qgfxraster_qws.cpp	Mon Oct 27 14:27:25 2003
@@ -75,7 +75,11 @@
 # if __BYTE_ORDER == __BIG_ENDIAN
 #  define QWS_BIG_ENDIAN
 # endif
+#elif defined(__APPLE__)
+#  define QWS_BIG_ENDIAN
 #endif
+// AJ: the above (test for __APPLE__) is very hacky, but I can't
+// think of something better at the moment.
 
 // Pull this private function in from qglobal.cpp
 extern unsigned int qt_int_sqrt( unsigned int n );
@@ -3770,6 +3774,19 @@
 	    b = srcval & 0xff;
 
 	    unsigned char * tmp=(unsigned char *)&alphabuf[loopc];
+# ifdef QWS_BIG_ENDIAN
+	    if(av==255) {
+	        // Do nothing - we already have source values in r,g,b
+	    } else if(av==0) {
+	        r = *(tmp+1);
+	        g = *(tmp+2);
+	        b = *(tmp+3);
+	    } else {
+		r = ((r-*(tmp+1)) * av) / 256 + *(tmp+1);
+		g = ((g-*(tmp+2)) * av) / 256 + *(tmp+2);
+		b = ((b-*(tmp+3)) * av) / 256 + *(tmp+3);
+	    }
+# else
 	    if(av==255) {
 	        // Do nothing - we already have source values in r,g,b
 	    } else if(av==0) {
@@ -3781,6 +3798,7 @@
 		g = ((g-*(tmp+1)) * av) / 256 + *(tmp+1);
 		b = ((b-*(tmp+0)) * av) / 256 + *(tmp+0);
 	    }
+# endif
 	    *(alphaptr++) = (r << 16) | (g << 8) | b;
 	}
 
@@ -4204,7 +4222,7 @@
 
     } else if ( depth == 1 ) {
 	if (srctype==SourceImage) {
-	    static int warn;
+	    static int warn = 0;
 	    if ( warn++ < 5 )
 		qDebug( "bitmap alpha-image not implemented" );
 	    hImageLineUnclipped( x1, x2, l, srcdata, FALSE );
@@ -4237,7 +4255,7 @@
     setAlphaType(IgnoreAlpha);
     if ( w <= 0 || h <= 0 || !ncliprect ) return;
     GFX_START(QRect(rx+xoffs, ry+yoffs, w+1, h+1))
-#ifdef QWS_EXPERIMENTAL_FASTPATH
+#if 0 // def QWS_EXPERIMENTAL_FASTPATH    !! this is crashing HancomWord on OZ !!
     // ### fix for 8bpp
     // This seems to be reliable now, at least for 16bpp
 
diff -uNr --exclude-from=./diff_ignore qt-2.3.7/src/kernel/qkeyboard_qws.cpp qte-2.3.7/src/kernel/qkeyboard_qws.cpp
--- qt-2.3.7/src/kernel/qkeyboard_qws.cpp	Thu Jul 17 03:20:25 2003
+++ qte-2.3.7/src/kernel/qkeyboard_qws.cpp	Mon Oct 27 14:42:11 2003
@@ -69,7 +69,7 @@
 #if !defined(_OS_QNX6_)
 
 #include <termios.h>
-#if !defined(_OS_FREEBSD_) && !defined(_OS_SOLARIS_)
+#if !defined(_OS_MACX_) && !defined(_OS_FREEBSD_) && !defined(_OS_SOLARIS_)
 #include <sys/kd.h>
 #include <sys/vt.h>
 #endif
@@ -468,10 +468,12 @@
 void QWSPC101KeyboardHandler::restoreLeds()
 {
     char leds;
+#if !defined(_OS_MACX_)
     ioctl(0, KDGETLED, &leds);
     leds = leds & ~LED_CAP;
     if ( caps ) leds |= LED_CAP;
     ioctl(0, KDSETLED, leds);
+#endif
 }
 
 class QWSTtyKeyboardHandler : public QWSPC101KeyboardHandler
@@ -547,7 +549,7 @@
 
 static void vtSwitchHandler(int /*sig*/)
 {
-#if !defined(_OS_FREEBSD_) && !defined(_OS_SOLARIS_)
+#if !defined(_OS_MACX_) && !defined(_OS_FREEBSD_) && !defined(_OS_SOLARIS_)
     if (vtActive) {
 	qwsServer->enablePainting(false);
 	qt_screen->save();
@@ -880,7 +882,7 @@
     if (term && !release) {
 	ctrl = false;
 	alt = false;
-#if !defined(_OS_FREEBSD_) && !defined(_OS_SOLARIS_)
+#if !defined(_OS_MACX_) && !defined(_OS_FREEBSD_) && !defined(_OS_SOLARIS_)
 	ioctl(kbdFD, VT_ACTIVATE, term);
 #endif
 	return;
@@ -1031,7 +1033,7 @@
 	struct termios termdata;
 	tcgetattr( kbdFD, &termdata );
 
-#if !defined(_OS_FREEBSD_) && !defined(_OS_SOLARIS_)
+#if !defined(_OS_MACX_) && !defined(_OS_FREEBSD_) && !defined(_OS_SOLARIS_)
 	ioctl(kbdFD, KDSKBMODE, K_RAW);
 #endif
 
@@ -1047,7 +1049,7 @@
 
 	signal(VTSWITCHSIG, vtSwitchHandler);
 
-#if !defined(_OS_FREEBSD_) && !defined(_OS_SOLARIS_)
+#if !defined(_OS_MACX_) && !defined(_OS_FREEBSD_) && !defined(_OS_SOLARIS_)
 	struct vt_mode vtMode;
 	ioctl(kbdFD, VT_GETMODE, &vtMode);
 
@@ -1068,7 +1070,7 @@
 {
     if (kbdFD >= 0)
     {
-#if !defined(_OS_FREEBSD_) && !defined(_OS_SOLARIS_)
+#if !defined(_OS_MACX_) && !defined(_OS_FREEBSD_) && !defined(_OS_SOLARIS_)
 	ioctl(kbdFD, KDSKBMODE, K_XLATE);
 #endif
 	tcsetattr(kbdFD, TCSANOW, &origTermData);
diff -uNr --exclude-from=./diff_ignore qt-2.3.7/src/kernel/qkeyboard_qws.cpp.rej qte-2.3.7/src/kernel/qkeyboard_qws.cpp.rej
--- qt-2.3.7/src/kernel/qkeyboard_qws.cpp.rej	Thu Jan  1 01:00:00 1970
+++ qte-2.3.7/src/kernel/qkeyboard_qws.cpp.rej	Mon Oct 27 14:27:25 2003
@@ -0,0 +1,34 @@
+***************
+*** 1023,1029 ****
+      struct termios termdata;
+      tcgetattr( kbdFD, &termdata );
+  
+- #if !defined(_OS_FREEBSD_) && !defined(_OS_SOLARIS_)
+      ioctl(kbdFD, KDSKBMODE, K_RAW);
+  #endif
+  
+--- 1023,1029 ----
+      struct termios termdata;
+      tcgetattr( kbdFD, &termdata );
+  
++ #if !defined(_OS_MACX_) && !defined(_OS_FREEBSD_) && !defined(_OS_SOLARIS_)
+      ioctl(kbdFD, KDSKBMODE, K_RAW);
+  #endif
+  
+***************
+*** 1039,1045 ****
+  
+      signal(VTSWITCHSIG, vtSwitchHandler);
+  
+- #if !defined(_OS_FREEBSD_) && !defined(_OS_SOLARIS_)
+      struct vt_mode vtMode;
+      ioctl(kbdFD, VT_GETMODE, &vtMode);
+  
+--- 1039,1045 ----
+  
+      signal(VTSWITCHSIG, vtSwitchHandler);
+  
++ #if !defined(_OS_MACX_) && !defined(_OS_FREEBSD_) && !defined(_OS_SOLARIS_)
+      struct vt_mode vtMode;
+      ioctl(kbdFD, VT_GETMODE, &vtMode);
+  
diff -uNr --exclude-from=./diff_ignore qt-2.3.7/src/kernel/qlock_qws.cpp qte-2.3.7/src/kernel/qlock_qws.cpp
--- qt-2.3.7/src/kernel/qlock_qws.cpp	Thu Jul 17 03:20:25 2003
+++ qte-2.3.7/src/kernel/qlock_qws.cpp	Mon Oct 27 14:27:25 2003
@@ -38,7 +38,7 @@
 #include <errno.h>
 
 #if defined(__GNU_LIBRARY__) && !defined(_SEM_SEMUN_UNDEFINED) \
-    || defined _OS_FREEBSD_
+    || defined(_OS_FREEBSD_) || defined(_OS_MACX_)
 /* union semun is defined by including <sys/sem.h> */
 #else
 /* according to X/OPEN we have to define it ourselves */
diff -uNr --exclude-from=./diff_ignore qt-2.3.7/src/kernel/qwindowsystem_qws.cpp qte-2.3.7/src/kernel/qwindowsystem_qws.cpp
--- qt-2.3.7/src/kernel/qwindowsystem_qws.cpp	Thu Jul 17 03:20:26 2003
+++ qte-2.3.7/src/kernel/qwindowsystem_qws.cpp	Mon Oct 27 14:27:25 2003
@@ -844,6 +844,18 @@
 {
 }
 
+static void catchSegvSignal( int )
+{
+#ifndef QT_NO_QWS_KEYBOARD
+    if ( qwsServer )
+	qwsServer->closeKeyboard();
+#endif
+    QWSServer::closedown();
+    fprintf(stderr, "Segmentation fault.\n");
+    exit(1);
+}
+
+
 /*!
   \class QWSServer qwindowsystem_qws.h
   \brief Server-specific functionality in Qt/Embedded
@@ -936,6 +948,7 @@
     }
 
     signal(SIGPIPE, ignoreSignal); //we get it when we read
+    signal(SIGSEGV, catchSegvSignal); //recover the keyboard on crash
 #endif
     focusw = 0;
     mouseGrabber = 0;
@@ -950,7 +963,7 @@
 
 #ifndef QT_NO_QWS_MULTIPROCESS
     if ( !geteuid() ) {
-#if !defined(_OS_FREEBSD_) && !defined(_OS_SOLARIS_)
+#if !defined(_OS_MACX_) && !defined(_OS_FREEBSD_) && !defined(_OS_SOLARIS_)
 	if( mount(0, "/var/shm", "shm", 0, 0) ) {
 	    /* This just confuses people with 2.2 kernels
 	    if ( errno != EBUSY )
diff -uNr --exclude-from=./diff_ignore qt-2.3.7/src/kernel/qwsmouse_qws.cpp qte-2.3.7/src/kernel/qwsmouse_qws.cpp
--- qt-2.3.7/src/kernel/qwsmouse_qws.cpp	Thu Jul 17 03:20:26 2003
+++ qte-2.3.7/src/kernel/qwsmouse_qws.cpp	Mon Oct 27 14:27:25 2003
@@ -336,7 +336,7 @@
 	tty.c_oflag     = 0;
 	tty.c_lflag     = 0;
 	tty.c_cflag     = f | CREAD | CLOCAL | HUPCL;
-#if !defined(_OS_FREEBSD_) && !defined(_OS_SOLARIS_)
+#if !defined(_OS_MACX_) && !defined(_OS_FREEBSD_) && !defined(_OS_SOLARIS_)
 	tty.c_line      = 0;
 #endif
 	tty.c_cc[VTIME] = 0;
@@ -889,7 +889,7 @@
 		tty.c_iflag = IGNBRK | IGNPAR;
 		tty.c_oflag = 0;
 		tty.c_lflag = 0;
-#if !defined(_OS_FREEBSD_) && !defined(_OS_SOLARIS_)
+#if !defined(_OS_MACX_) && !defined(_OS_FREEBSD_) && !defined(_OS_SOLARIS_)
 		tty.c_line = 0;
 #endif // _OS_FREEBSD_
 		tty.c_cc[VTIME] = 0;
Binary files qt-2.3.7/src/moc/moc and qte-2.3.7/src/moc/moc differ
diff -uNr --exclude-from=./diff_ignore qt-2.3.7/src/network/qdns.cpp qte-2.3.7/src/network/qdns.cpp
--- qt-2.3.7/src/network/qdns.cpp	Thu Jul 17 03:20:26 2003
+++ qte-2.3.7/src/network/qdns.cpp	Sun Nov 16 13:44:27 2003
@@ -1991,7 +1991,7 @@
     domains = new QStrList( TRUE );
     domains->setAutoDelete( TRUE );
 
-    res_init();
+    // res_init(); (eilers: Caused link problems)
     int i;
     // find the name servers to use
     for( i=0; i < MAXNS && i < _res.nscount; i++ ) {
diff -uNr --exclude-from=./diff_ignore qt-2.3.7/src/tools/qconfig-qpe.h qte-2.3.7/src/tools/qconfig-qpe.h
--- qt-2.3.7/src/tools/qconfig-qpe.h	Thu Jan  1 01:00:00 1970
+++ qte-2.3.7/src/tools/qconfig-qpe.h	Sun Dec 14 12:48:43 2003
@@ -0,0 +1,100 @@
+/**********************************************************************
+** Copyright (C) 2000 Trolltech AS.  All rights reserved.
+**
+** This file is part of Qtopia Environment.
+**
+** This file may be distributed and/or modified under the terms of the
+** GNU General Public License version 2 as published by the Free Software
+** Foundation and appearing in the file LICENSE.GPL included in the
+** packaging of this file.
+**
+** This file is provided AS IS with NO WARRANTY OF ANY KIND, INCLUDING THE
+** WARRANTY OF DESIGN, MERCHANTABILITY AND FITNESS FOR A PARTICULAR PURPOSE.
+**
+** See http://www.trolltech.com/gpl/ for GPL licensing information.
+**
+** Contact info@trolltech.com if any conditions of this licensing are
+** not clear to you.
+**
+**********************************************************************/
+#ifndef QT_H
+#endif // QT_H
+
+// Empty leaves all features enabled.  See doc/html/features.html for choices.
+
+// Note that disabling some features will produce a libqt that is not
+// compatible with other libqt builds. Such modifications are only
+// supported on Qt/Embedded where reducing the library size is important
+// and where the application-suite is often a fixed set.
+
+#ifndef QT_DLL
+#define QT_DLL // Internal
+#endif
+
+#if defined(QT_QWS_IPAQ) || defined(QT_QWS_CASSIOPEIA) || defined(QT_QWS_SL5XXX)
+# define QT_NO_QWS_CURSOR
+# define QT_NO_QWS_MOUSE_AUTO
+#endif
+#ifndef QT_NO_CODECS
+#define QT_NO_CODECS
+#endif
+#define QT_NO_UNICODETABLES
+//#define QT_NO_IMAGEIO_BMP
+#define QT_NO_IMAGEIO_PPM
+//#define QT_NO_ASYNC_IO
+//#define QT_NO_ASYNC_IMAGE_IO
+#define QT_NO_FREETYPE
+#define QT_NO_BDF
+//#define QT_NO_FONTDATABASE
+#define QT_NO_DRAGANDDROP
+//#define QT_NO_CLIPBOARD
+#define QT_NO_PROPERTIES
+#define QT_NO_NETWORKPROTOCOL
+
+#define QT_NO_IMAGE_TEXT
+
+//#define QT_NO_TOOLTIP
+#define QT_NO_COLORNAMES
+#define QT_NO_TRANSFORMATIONS
+#define QT_NO_TRANSLATION_BUILDER
+#define QT_NO_COMPLEXTEXT
+#define QT_NO_PRINTER
+#define QT_NO_PICTURE
+//#define QT_NO_ICONVIEW
+#define QT_NO_DIAL
+#define QT_NO_SIZEGRIP
+#define QT_NO_WORKSPACE
+//#define QT_NO_TABLE
+//#define QT_NO_ACTION
+//#define QT_NO_SETTINGS
+#define QT_NO_STYLE_POCKETPC
+#ifndef QT_NO_STYLE_AQUA
+# define QT_NO_STYLE_AQUA
+#endif
+#define QT_NO_STYLE_MOTIF
+#define QT_NO_STYLE_PLATINUM
+#define QT_NO_FILEDIALOG
+#define QT_NO_FONTDIALOG
+#define QT_NO_PRINTDIALOG
+#define QT_NO_COLORDIALOG
+#define QT_NO_INPUTDIALOG
+//#define QT_NO_MESSAGEBOX
+#define QT_NO_PROGRESSDIALOG
+//#define QT_NO_TABDIALOG
+#define QT_NO_WIZARD
+#define QT_NO_EFFECTS
+//#define QT_NO_COMPONENT
+#define QT_NO_DOM
+#define QT_NO_SEMIMODAL
+//#define QT_NO_PROGRESSBAR
+#define QT_NO_SPLITTER
+
+//#define QT_NO_QWS_SAVEFONTS
+//#define QT_NO_QWS_PROPERTIES
+
+#define QT_NO_QWS_BEOS_WM_STYLE
+#define QT_NO_QWS_KDE2_WM_STYLE
+#define QT_NO_QWS_KDE_WM_STYLE
+#define QT_NO_QWS_WINDOWS_WM_STYLE
+
+#define QT_NO_SOUND
diff -uNr --exclude-from=./diff_ignore qt-2.3.7/src/tools/qconfig.h qte-2.3.7/src/tools/qconfig.h
--- qt-2.3.7/src/tools/qconfig.h	Fri Jan 26 14:43:06 2001
+++ qte-2.3.7/src/tools/qconfig.h	Thu Dec 11 12:54:20 2003
@@ -1,3 +1,26 @@
+// Copied from qconfig-qpe.h
+/**********************************************************************
+** Copyright (C) 2000 Trolltech AS.  All rights reserved.
+**
+** This file is part of Qtopia Environment.
+**
+** This file may be distributed and/or modified under the terms of the
+** GNU General Public License version 2 as published by the Free Software
+** Foundation and appearing in the file LICENSE.GPL included in the
+** packaging of this file.
+**
+** This file is provided AS IS with NO WARRANTY OF ANY KIND, INCLUDING THE
+** WARRANTY OF DESIGN, MERCHANTABILITY AND FITNESS FOR A PARTICULAR PURPOSE.
+**
+** See http://www.trolltech.com/gpl/ for GPL licensing information.
+**
+** Contact info@trolltech.com if any conditions of this licensing are
+** not clear to you.
+**
+**********************************************************************/
+#ifndef QT_H
+#endif // QT_H
+
 // Empty leaves all features enabled.  See doc/html/features.html for choices.
 
 // Note that disabling some features will produce a libqt that is not
@@ -8,3 +31,71 @@
 #ifndef QT_DLL
 #define QT_DLL // Internal
 #endif
+
+#if defined(QT_QWS_IPAQ) || defined(QT_QWS_CASSIOPEIA) || defined(QT_QWS_SL5XXX)
+# define QT_NO_QWS_CURSOR
+# define QT_NO_QWS_MOUSE_AUTO
+#endif
+#ifndef QT_NO_CODECS
+#define QT_NO_CODECS
+#endif
+#define QT_NO_UNICODETABLES
+//#define QT_NO_IMAGEIO_BMP
+#define QT_NO_IMAGEIO_PPM
+//#define QT_NO_ASYNC_IO
+//#define QT_NO_ASYNC_IMAGE_IO
+#define QT_NO_FREETYPE
+#define QT_NO_BDF
+//#define QT_NO_FONTDATABASE
+#define QT_NO_DRAGANDDROP
+//#define QT_NO_CLIPBOARD
+#define QT_NO_PROPERTIES
+#define QT_NO_NETWORKPROTOCOL
+
+#define QT_NO_IMAGE_TEXT
+
+//#define QT_NO_TOOLTIP
+#define QT_NO_COLORNAMES
+#define QT_NO_TRANSFORMATIONS
+#define QT_NO_TRANSLATION_BUILDER
+#define QT_NO_COMPLEXTEXT
+#define QT_NO_PRINTER
+#define QT_NO_PICTURE
+//#define QT_NO_ICONVIEW
+#define QT_NO_DIAL
+#define QT_NO_SIZEGRIP
+#define QT_NO_WORKSPACE
+//#define QT_NO_TABLE
+//#define QT_NO_ACTION
+//#define QT_NO_SETTINGS
+#define QT_NO_STYLE_POCKETPC
+#ifndef QT_NO_STYLE_AQUA
+# define QT_NO_STYLE_AQUA
+#endif
+#define QT_NO_STYLE_MOTIF
+#define QT_NO_STYLE_PLATINUM
+#define QT_NO_FILEDIALOG
+#define QT_NO_FONTDIALOG
+#define QT_NO_PRINTDIALOG
+#define QT_NO_COLORDIALOG
+#define QT_NO_INPUTDIALOG
+//#define QT_NO_MESSAGEBOX
+#define QT_NO_PROGRESSDIALOG
+//#define QT_NO_TABDIALOG
+#define QT_NO_WIZARD
+#define QT_NO_EFFECTS
+//#define QT_NO_COMPONENT
+#define QT_NO_DOM
+#define QT_NO_SEMIMODAL
+//#define QT_NO_PROGRESSBAR
+#define QT_NO_SPLITTER
+
+//#define QT_NO_QWS_SAVEFONTS
+//#define QT_NO_QWS_PROPERTIES
+
+#define QT_NO_QWS_BEOS_WM_STYLE
+#define QT_NO_QWS_KDE2_WM_STYLE
+#define QT_NO_QWS_KDE_WM_STYLE
+#define QT_NO_QWS_WINDOWS_WM_STYLE
+
+#define QT_NO_SOUND
\ No newline at end of file
diff -uNr --exclude-from=./diff_ignore qt-2.3.7/src/tools/qmodules.h qte-2.3.7/src/tools/qmodules.h
--- qt-2.3.7/src/tools/qmodules.h	Fri Aug 31 17:00:56 2001
+++ qte-2.3.7/src/tools/qmodules.h	Thu Dec 11 12:53:57 2003
@@ -1,14 +1,12 @@
-#ifndef QT_H
-#endif // QT_H
-
+// These modules are licensed to you
 #define QT_MODULE_TOOLS
 #define QT_MODULE_KERNEL
 #define QT_MODULE_WIDGETS
 #define QT_MODULE_DIALOGS
-
 #define QT_MODULE_ICONVIEW
 #define QT_MODULE_WORKSPACE
-#define QT_MODULE_TABLE
+#define QT_MODULE_NETWORK
 #define QT_MODULE_CANVAS
+#define QT_MODULE_TABLE
 #define QT_MODULE_XML
-#define QT_MODULE_NETWORK
+#define QT_MODULE_OPENGL
diff -uNr --exclude-from=./diff_ignore qt-2.3.7/src/widgets/qcommonstyle.cpp qte-2.3.7/src/widgets/qcommonstyle.cpp
--- qt-2.3.7/src/widgets/qcommonstyle.cpp	Thu Jul 17 03:20:26 2003
+++ qte-2.3.7/src/widgets/qcommonstyle.cpp	Mon Oct 27 14:21:28 2003
@@ -566,7 +566,7 @@
 				    bool enabled, bool active )
 {
 #ifndef QT_NO_MENUBAR
-#ifndef QT_NO_STYLE_SGI
+#if 1 // #ifndef QT_NO_STYLE_SGI
     if (draw_menu_bar_impl != 0) {
 	QDrawMenuBarItemImpl impl = draw_menu_bar_impl;
 	(this->*impl)(p, x, y, w, h, mi, g, enabled, active);
diff -uNr --exclude-from=./diff_ignore qt-2.3.7/src/widgets/qlistview.cpp qte-2.3.7/src/widgets/qlistview.cpp
--- qt-2.3.7/src/widgets/qlistview.cpp	Thu Jul 17 03:20:26 2003
+++ qte-2.3.7/src/widgets/qlistview.cpp	Mon Oct 27 14:21:28 2003
@@ -4968,9 +4968,9 @@
 	l = l->childItem ? l->childItem : l->siblingItem;
 
     if ( l && l->height() )
-	s.setHeight( s.height() + 10 * l->height() );
-    else
-	s.setHeight( s.height() + 140 );
+	s.setHeight( s.height() + 4 /*10*/ * l->height() );
+    else                      // ^v much too big for handhelds
+	s.setHeight( s.height() + 30 /*140*/ );
 
     if ( s.width() > s.height() * 3 )
 	s.setHeight( s.width() / 3 );
diff -uNr --exclude-from=./diff_ignore qt-2.3.7/src/widgets/qtoolbutton.cpp qte-2.3.7/src/widgets/qtoolbutton.cpp
--- qt-2.3.7/src/widgets/qtoolbutton.cpp	Thu Jul 17 03:20:27 2003
+++ qte-2.3.7/src/widgets/qtoolbutton.cpp	Mon Oct 27 14:21:28 2003
@@ -230,7 +230,7 @@
 	else
 	    QToolTip::add( this, textLabel );
     }
-#endif	
+#endif
 }
 
 
@@ -324,12 +324,12 @@
 	QPixmap pm = iconSet(TRUE).pixmap(QIconSet::Large, QIconSet::Normal);
 	w = pm.width();
 	h = pm.height();
-	if ( w < 32 )
-	    w = 32;
-	if ( h < 32 )
-	    h = 32;
+	if ( w < 24 )
+	    w = 24;
+	if ( h < 24 )
+	    h = 24;
     } else {
-	w = h = 16;
+	w = h = 14;
 	QPixmap pm = iconSet(TRUE).pixmap(QIconSet::Small, QIconSet::Normal);
 	w = pm.width();
 	h = pm.height();
diff -uNr --exclude-from=./diff_ignore qt-2.3.7/src-mt.mk qte-2.3.7/src-mt.mk
--- qt-2.3.7/src-mt.mk	Thu Jan  1 01:00:00 1970
+++ qte-2.3.7/src-mt.mk	Thu Dec 11 12:53:57 2003
@@ -0,0 +1,2 @@
+all:
+	@echo "(not building threaded Qt)"
